# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: tentative/status.sub.window.js
# Commit: 680c502179c6
# Full Hash: 680c502179c6a98e8e6960fbab016d3aea44ff4c
# Author: Andreas Farre <farre@mozilla.com>
# Date: 2023-05-10 21:37:01
# Regressor Bug: 1823877
# File Overlap Count: 1
# Description:
#   Bug 1823877 - Part 1: Filter opaque results from fetch() in the parent for ORB. r=sefeng,smaug,necko-reviewers,edenchuang,valentin
#   
#   We make sure to not send any data to the content process in case of
#   fetching an opaque resource. This is way to remain more web
#   compatible, but is also in conflict with the ORB specification.
# ==============================================================================

diff -r 8295c1c7bfa0 -r 680c502179c6 testing/web-platform/mozilla/tests/fetch/orb/tentative/status.sub.window.js
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/testing/web-platform/mozilla/tests/fetch/orb/tentative/status.sub.window.js	Wed May 10 14:35:52 2023 +0000
@@ -0,0 +1,30 @@
+// META: script=/fetch/orb/resources/utils.js
+// META: script=resources/utils.js
+
+const path = "http://{{domains[www1]}}:{{ports[http][0]}}/fetch/orb/resources";
+
+// Due to web compatibility we filter opaque Response object from the
+// fetch() function in the Fetch specification. See Bug 1823877. This
+// might be removed in the future.
+promise_internal_response_is_filtered(
+  fetchORB(
+    `${path}/data.json`,
+    null,
+    contentType("application/json"),
+    "status(206)"
+  ),
+  "ORB should filter opaque-response-blocklisted MIME type with status 206"
+);
+
+// Due to web compatibility we filter opaque Response object from the
+// fetch() function in the Fetch specification. See Bug 1823877. This
+// might be removed in the future.
+promise_internal_response_is_filtered(
+  fetchORB(
+    `${path}/data.json`,
+    null,
+    contentType("application/json"),
+    "status(302)"
+  ),
+  "ORB should filter opaque range of image/png not starting at zero, that isn't subsequent"
+);
